### DISCUSSION TITLE
Inverse problem with "hidden" unknown parameter

### DISCUSSION THREAD

**Discussion by gcappellini** [lululxvi/deepxde#973] [Q&A]
Dear all,

I am trying to implement a Time-dependent PDE inverse problem, where the unknown parameter doesn't appear in the PDE formulation, but in a previous step.
 
The equation is the [Okajima's adimensionalized version of Pennes' Bioheat Eq](https://www.sciencedirect.com/science/article/pii/S0306456509000825) (eq. 22 with n=0).
I want to import a dataset of thermal measurements, then adimensionalize it using certain Ci coefficients that depend on the variable ωb.
The thermal measurements dataset is generated from a similar Time-dependent **(direct)** PDE problem, with a constant, arbitrary value of ωb.
In the inverse problem, the variable  ωb is initialized with a different value. 
I want the network to recover the arbitrary value imposed in the direct problem for the generation of thermal measurements, but when I run the code, the variable value doesn't change.

[Here](https://colab.research.google.com/drive/1Q7bzvyiO825SPbn9-rx3rYuWo19clEJG#scrollTo=zUauWpAgyFMc) you can find the code and the pretrained direct model.
Is there a way for the gradient to reach ωb only, or do I have to initialize other variables?

Hope it is all clear.

Thanks,
Guglielmo

**Comment by lululxvi** [comment]
Can you use a simple example to demonstrate your question "the unknown parameter doesn't appear in the PDE formulation, but in a previous step."?

**Reply by gcappellini** [reply]
Here attached you can find a simple example that I wrote starting from [this problem](https://github.com/lululxvi/deepxde/blob/master/examples/pinn_inverse/elliptic_inverse_field_batch.py).
I would like the network to recover the original value for the variable, i.e., C=1.
You can find the same example on a colab file [here](https://colab.research.google.com/drive/1aimnvnGHVzW6kF033MqqEu19ePnOw6In?usp=sharing).

```
import deepxde as dde
import numpy as np

C=dde.Variable(1.3)

def gen_traindata(num):
    # generate num equally-spaced points from -1 to 1
    xvals = np.linspace(-1, 1, num).reshape(num, 1)
    uvals = C.cpu().detach().numpy()*np.sin(np.pi * xvals)
    return xvals, uvals


def pde(x, y):
    u, q = y[:, 0:1], y[:, 1:2]
    du_xx = dde.grad.hessian(y, x, component=0, i=0, j=0)
    return -du_xx + q


def sol(x):
    # solution is u(x) = sin(pi*x), q(x) = -pi^2 * sin(pi*x)
    return np.sin(np.pi * x)


geom = dde.geometry.Interval(-1, 1)

bc = dde.icbc.DirichletBC(geom, sol, lambda _, on_boundary: on_boundary, component=0)
ob_x, ob_u = gen_traindata(10000)
observe_u = dde.icbc.PointSetBC(ob_x, ob_u, component=0, batch_size=100)

data = dde.data.PDE(
    geom,
    pde,
    [bc, observe_u],
    num_domain=200,
    num_boundary=2,
    num_test=1000,
    anchors=ob_x
)

net = dde.nn.PFNN([1, [20, 20], [20, 20], [20, 20], 2], "tanh", "Glorot uniform")

model = dde.Model(data, net)
model.compile("adam", lr=0.0001, loss_weights=[1, 100, 1000], external_trainable_variables=C)
variable = dde.callbacks.VariableValue(C, period=1000)
losshistory, train_state = model.train(iterations=5000, callbacks=[variable])
dde.saveplot(losshistory, train_state, issave=True, isplot=True)

```

**Reply by rodrigogdourado** [reply]
> Here attached you can find a simple example that I wrote starting from [this problem](https://github.com/lululxvi/deepxde/blob/master/examples/pinn_inverse/elliptic_inverse_field_batch.py). I would like the network to recover the original value for the variable, i.e., C=1. You can find the same example on a colab file [here](https://colab.research.google.com/drive/1aimnvnGHVzW6kF033MqqEu19ePnOw6In?usp=sharing).
> 
> ```
> import deepxde as dde
> import numpy as np
> 
> C=dde.Variable(1.3)
> 
> def gen_traindata(num):
>     # generate num equally-spaced points from -1 to 1
>     xvals = np.linspace(-1, 1, num).reshape(num, 1)
>     uvals = C.cpu().detach().numpy()*np.sin(np.pi * xvals)
>     return xvals, uvals
> 
> 
> def pde(x, y):
>     u, q = y[:, 0:1], y[:, 1:2]
>     du_xx = dde.grad.hessian(y, x, component=0, i=0, j=0)
>     return -du_xx + q
> 
> 
> def sol(x):
>     # solution is u(x) = sin(pi*x), q(x) = -pi^2 * sin(pi*x)
>     return np.sin(np.pi * x)
> 
> 
> geom = dde.geometry.Interval(-1, 1)
> 
> bc = dde.icbc.DirichletBC(geom, sol, lambda _, on_boundary: on_boundary, component=0)
> ob_x, ob_u = gen_traindata(10000)
> observe_u = dde.icbc.PointSetBC(ob_x, ob_u, component=0, batch_size=100)
> 
> data = dde.data.PDE(
>     geom,
>     pde,
>     [bc, observe_u],
>     num_domain=200,
>     num_boundary=2,
>     num_test=1000,
>     anchors=ob_x
> )
> 
> net = dde.nn.PFNN([1, [20, 20], [20, 20], [20, 20], 2], "tanh", "Glorot uniform")
> 
> model = dde.Model(data, net)
> model.compile("adam", lr=0.0001, loss_weights=[1, 100, 1000], external_trainable_variables=C)
> variable = dde.callbacks.VariableValue(C, period=1000)
> losshistory, train_state = model.train(iterations=5000, callbacks=[variable])
> dde.saveplot(losshistory, train_state, issave=True, isplot=True)
> ```

In general, when we solve an inverse problem we try to estimate an unknown parameter (or function) of the boundary conditions, initial condition or of the PDE using know data at some points within the domain. If I understand it correctly, in your case, you are trying to estimate an unknown parameter in the data used as reference to solve the inverse problem, which doesn't make sense to me.
